modules/charging.c

```diff
@@ -366,6 +394,13 @@ mch_policy_evaluate_charging_state(void)
 
     if( usb_cable_state == USB_CABLE_DISCONNECTED ) {
         switch( mch_charging_mode ) {
+        case CHARGING_MODE_APPLY_THRESHOLDS_AFTER_FULL:
+            if ( mch_charging_mode_evaluated_with_cable_connected ) {
+                mce_setting_set_int(MCE_SETTING_CHARGING_MODE,
```

Comment:
This looks suspicious (from mce pov settings are read-only - apart from sanitizing changes made by client side)

Reply:
yeah.. when you think of the use case more generally, it is like user scheduling an action by raising a flag and the flag needs to be dropped when it happens. We have two options: either introduce a completely new mechanism (from persistence to UI) or use settings for that. What would you choose?

Reply:
I really dislike that policy mode is changed here as it completely breaks existing functionality that I happen to use and rely on...

A live and let live solution could be: use separate boolean override for this new "charge once" behavior, detach settings ui from underlying reality a bit more than what it already is (say, by treating apply-thresholds-after-full as it were apply-thresholds) -> command line can still be used for tweaking by people not served by ui.